### DISCUSSION TITLE
Link to the same page not reloading the page

### DISCUSSION THREAD

**Discussion by phstc** [vercel/next.js#63926] [App Router]
I have an app with a sidebar; I have a few menu items on that sidebar. Let's say there's a menu item called People:

```
[menu]
...
People
...
[/menu]
```

If I click on People, it opens `/people`, where I can filter people by name. This changes the URL to `/people?name=...` and lists filtered people by name. 

**The issue:** I'm currently on `/people?name=...`, and then I click again on the People link on the sidebar to load `/people`. Next.js `Link` changes the URL from  `/people?name=...` to `/people` but does not update the page, so the page is still filtering by name. 

Replacing `Link` with `a` works, but then it will reload the whole page. 

Any other suggestions?

**Comment by shalldie** [comment]
I see `14.2.5` solved this problem. https://github.com/vercel/next.js/pull/65977
